admin_auto_filters/filters.py

```diff
@@ -1,35 +1,118 @@
+from django.contrib.admin.utils import prepare_lookup_value
+from django.contrib.admin.widgets import (
+    AutocompleteSelect as BaseAutocompleteSelect,
+    AutocompleteSelectMultiple as BaseAutocompleteSelectMultiple,
+)
+from django.core.exceptions import ImproperlyConfigured
 from django.db.models import ForeignObjectRel
 from django.db.models.constants import LOOKUP_SEP  # this is '__'
-from django.db.models.fields.related_descriptors import ReverseManyToOneDescriptor, ManyToManyDescriptor
+from django.db.models.fields.related_descriptors import (
+    ReverseManyToOneDescriptor, ManyToManyDescriptor,
+)
+from django.forms import ModelChoiceField, ModelMultipleChoiceField
 from django.forms.widgets import Media, MEDIA_TYPES, media_property
 from django.shortcuts import reverse
 
 
-class AutocompleteSelect(Base):
+class AutocompleteSelect(BaseAutocompleteSelect):
+    """A customize AutocompleteSelect that allows a custom URL."""
+
     def __init__(self, rel, admin_site, attrs=None, choices=(), using=None, custom_url=None):
+        """Initialize class variables for an AutocompleteSelect object."""
         self.custom_url = custom_url
         super().__init__(rel, admin_site, attrs, choices, using)
     
     def get_url(self):
+        """Specifies the URL to be used to fetch the autocomplete list."""
+        return self.custom_url if self.custom_url else super().get_url()
+
+
+class AutocompleteSelectMultiple(BaseAutocompleteSelectMultiple):
+    """A customize AutocompleteSelectMultiple that allows a custom URL."""
+
+    def __init__(self, rel, admin_site, attrs=None, choices=(), using=None, custom_url=None):
+        """Initialize class variables for an AutocompleteSelectMultiple object."""
+        self.custom_url = custom_url
+        super().__init__(rel, admin_site, attrs, choices, using)
+
+    def get_url(self):
+        """Specifies the URL to be used to fetch the autocomplete list."""
         return self.custom_url if self.custom_url else super().get_url()
 
 
-class AutocompleteFilter(admin.SimpleListFilter):
-    template = 'django-admin-autocomplete-filter/autocomplete-filter.html'
-    title = ''
-    field_name = ''
+class AutocompleteFilterMeta(type(SimpleListFilter)):
+    """
+    A metaclass for setting class-level variables expected by SimpleListFilter.
+    When Python 3.5 support is no longer required, this could possibly be
+    replaced by a parent class using __init_subclass__, rather than a metaclass.
+    """
+    def __init__(self, *args, **kwargs):
+        """
+        Initialize class variables for class `self`, for
+        subclasses of AutocompleteFilter.
+        """
+        super().__init__(*args, **kwargs)
+        if self.field_name is not None:
+            self.parameter_name = self.get_parameter_name()
+            # self.template =  # Default set in class body
+            self.title = self.get_title()
+
+
+class AutocompleteFilter(SimpleListFilter, metaclass=AutocompleteFilterMeta):
```

Comment:
I wonder if it is good that this class handle single autocomplete and autocomplete multiple. It seems too much...